src/mixins/template-render.js

```diff
@@ -32,9 +32,11 @@ export default {
   // object literal, or a function that returns an object
   // literal. All methods and attributes from this object
   // are copies to the object passed in.
-  mixinTemplateContext(target) {
+  mixinTemplateContext(serializedData) {
     const templateContext = _.result(this, 'templateContext');
-    return _.extend({}, target, templateContext);
+    if (!serializedData) { return templateContext; };
+    if (!templateContext) { return serializedData; };
```

Comment:
I would swar the check order since the most common situation is `!templateContext`